Lib/zipimport.py

```diff
@@ -321,6 +321,16 @@ def get_resource_reader(self, fullname):
         return ZipReader(self, fullname)
 
 
+    def invalidate_caches(self):
+        """Reload the file data of the archive path."""
+        try:
+            self._files = _read_directory(self.archive)
```

Comment:
Thanks, @miss-islington. I do appreciate this fix.

Just dropping a comment in case the impact of this was missed during the review.
This might impact the performance notably. For example, I only import two zipped packages `py4j` and `pyspark` but it slows down the speed of `importlib.invalidate_caches` up to 3500% because now it has to go through the zipped package.